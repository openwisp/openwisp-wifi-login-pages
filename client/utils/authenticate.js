import {sessionStorage} from "./storage";

const authenticate = (cookies, orgSlug) => {
  const token = cookies.get(`${orgSlug}_auth_token`);
  const sessionKey = sessionStorage.getItem(`${orgSlug}_auth_token`);
  if (sessionKey) {
    cookies.remove(`${orgSlug}_auth_token`, {path: "/"});
    cookies.remove(`${orgSlug}_username`, {path: "/"});
    return true;
  }
  if (token) return true;
  return false;
};
export default authenticate;
