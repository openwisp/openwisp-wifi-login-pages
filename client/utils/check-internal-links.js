const isInternalLink = (link) => {
  if (link.length && link[0] === "/") return true;
  return false;
};
export default isInternalLink;
