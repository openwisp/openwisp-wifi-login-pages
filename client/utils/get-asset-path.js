const getAssetPath = (slug, relativePath) => `/assets/${slug}/${relativePath}`;
export default getAssetPath;
