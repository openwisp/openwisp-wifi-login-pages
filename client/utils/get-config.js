const loadConfig = (slug) => import(`../configs/${slug}.json`);
const getConfig = async (slug) => {
  let config;
  try {
    config = await loadConfig(slug);
  } catch (err) {
    console.error(`Configuration for organization ${slug} does not exists.`);
    config = undefined;
  }
  return config;
};

export default getConfig;
