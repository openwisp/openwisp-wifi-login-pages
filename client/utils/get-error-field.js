import React from "react";

const getErrorField = (errors, key, nonField = false, className = "") =>
  errors[key] ? (
    <div className={!nonField ? `error ${className}` : "error non-field"}>
      <span className="icon">!</span>
      <span className="text">{errors[key]}</span>
    </div>
  ) : null;

export default getErrorField;
