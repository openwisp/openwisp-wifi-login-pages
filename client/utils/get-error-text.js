const getErrorText = (error, fallbackVal = "") => {
  const {data} = error.response;
  // eslint-disable-next-line no-nested-ternary
  return data.detail
    ? data.detail
    : data.non_field_errors
    ? data.non_field_errors[0]
    : fallbackVal;
};
export default getErrorText;
