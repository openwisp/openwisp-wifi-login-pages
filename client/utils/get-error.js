import React from "react";

const getError = (errors, key = null) => {
  const errorKey = key === null ? "nonField" : key;
  return errors[errorKey] ? (
    <div className={key !== null ? `error ${key}` : "error non-field"}>
      <span className="icon" />
      <span className="text">{errors[errorKey]}</span>
    </div>
  ) : null;
};

export default getError;
