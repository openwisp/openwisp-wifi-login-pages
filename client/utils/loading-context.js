import React from "react";

export const loadingContextValue = {setLoading: () => {}, getLoading: () => {}};
const LoadingContext = React.createContext(loadingContextValue);

export default LoadingContext;
