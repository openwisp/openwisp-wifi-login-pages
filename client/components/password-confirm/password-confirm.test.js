/* eslint-disable prefer-promise-reject-errors */
/* eslint-disable camelcase */
import axios from "axios";
import {shallow} from "enzyme";
import React from "react";
import PropTypes from "prop-types";
import {BrowserRouter as Router} from "react-router-dom";
import {Provider} from "react-redux";
import renderer from "react-test-renderer";
import {toast} from "react-toastify";
import {loadingContextValue} from "../../utils/loading-context";
import getConfig from "../../utils/get-config";
import loadTranslation from "../../utils/load-translation";
import PasswordConfirm from "./password-confirm";
import translation from "../../test-translation.json";
import PasswordToggleIcon from "../../utils/password-toggle";

jest.mock("axios");
jest.mock("../../utils/get-config");
jest.mock("../../utils/load-translation");
const defaultConfig = getConfig("default", true);
const createTestProps = (props) => ({
  language: "en",
  orgSlug: "default",
  orgName: "default name",
  configuration: defaultConfig,
  passwordConfirm: defaultConfig.components.password_reset_confirm_form,
  setTitle: jest.fn(),
  match: {
    params: {
      uid: "testUid",
      token: "testToken",
    },
  },
  ...props,
});

const getTranslationString = (msgid) => {
  try {
    return translation.translations[""][msgid].msgstr[0];
  } catch (err) {
    // eslint-disable-next-line no-console
    console.error(err, msgid);
    return msgid;
  }
};

describe("<PasswordConfirm /> rendering with placeholder translation tags", () => {
  const props = createTestProps();
  it("should render translation placeholder correctly", () => {
    const wrapper = shallow(<PasswordConfirm {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe("<PasswordConfirm /> rendering", () => {
  let props;
  let wrapper;

  beforeEach(() => {
    props = createTestProps();
    loadTranslation("en", "default");
    wrapper = shallow(<PasswordConfirm {...props} />);
  });

  it("should render correctly", () => {
    props = createTestProps();
    const mockedStore = {
      subscribe: () => {},
      dispatch: () => {},
      // needed to render <Contact/>
      getState: () => ({
        organization: {
          configuration: props.configuration,
        },
        language: props.language,
      }),
    };
    const component = renderer
      .create(
        <Provider store={mockedStore}>
          <Router>
            <PasswordConfirm {...props} />
          </Router>
        </Provider>,
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it("should render 2 input fields", () => {
    expect(wrapper.find(".input")).toHaveLength(2);
  });

  it("should render password field correctly", () => {
    expect(wrapper.find(".row.password label").text()).toBe(
      getTranslationString("PWD_LBL"),
    );
    const passwordInput = wrapper.find(".row.password input");
    expect(passwordInput.prop("placeholder")).toBe(
      getTranslationString("PWD_PHOLD"),
    );
    expect(passwordInput.prop("title")).toBe(
      getTranslationString("PWD_PTRN_DESC"),
    );
    expect(passwordInput.prop("type")).toBe("password");
  });

  it("should render password confirm field correctly", () => {
    expect(wrapper.find(".row.password-confirm label").text()).toBe(
      getTranslationString("CONFIRM_PWD_LBL"),
    );
    const confirmInput = wrapper.find(".row.password-confirm input");
    expect(confirmInput.prop("placeholder")).toBe(
      getTranslationString("CONFIRM_PWD_PHOLD"),
    );
    expect(confirmInput.prop("title")).toBe(
      getTranslationString("PWD_PTRN_DESC"),
    );
    expect(confirmInput.prop("type")).toBe("password");
  });
});

describe("<PasswordConfirm /> interactions", () => {
  let props;
  let wrapper;
  let originalError;
  let lastConsoleOutuput;

  beforeEach(() => {
    originalError = console.error;
    lastConsoleOutuput = null;
    console.error = (data) => {
      lastConsoleOutuput = data;
    };
    PasswordConfirm.contextTypes = {
      setLoading: PropTypes.func,
      getLoading: PropTypes.func,
    };
    props = createTestProps();
    wrapper = shallow(<PasswordConfirm {...props} />, {
      context: loadingContextValue,
    });
  });

  afterEach(() => {
    console.error = originalError;
  });

  it("should change state values when handleChange function is invoked", () => {
    wrapper
      .find(".password input")
      .simulate("change", {target: {value: "123456", name: "newPassword1"}});
    expect(wrapper.state("newPassword1")).toEqual("123456");
    wrapper
      .find(".password-confirm input")
      .simulate("change", {target: {value: "123456", name: "newPassword2"}});
    expect(wrapper.state("newPassword2")).toEqual("123456");
  });

  it("should execute handleSubmit correctly when form is submitted", () => {
    axios
      .mockImplementationOnce(() =>
        Promise.reject({response: {data: {detail: "errors"}}}),
      )
      .mockImplementationOnce(() =>
        Promise.reject({
          response: {data: {non_field_errors: ["non field errors"]}},
        }),
      )
      .mockImplementationOnce(() =>
        Promise.reject({
          response: {data: {token: ["Invalid token"]}},
        }),
      )
      .mockImplementationOnce(() => Promise.resolve({data: {detail: true}}));
    wrapper.setState({
      newPassword1: "wrong password",
      newPassword2: "wrong password1",
    });
    wrapper.instance().handleSubmit({preventDefault: () => {}});
    expect(wrapper.update().find(".password-confirm div.error")).toHaveLength(
      1,
    );
    wrapper.setState({
      newPassword1: "password",
      newPassword2: "password",
    });
    const spyToastError = jest.spyOn(toast, "error");
    const spyToastSuccess = jest.spyOn(toast, "success");
    return wrapper
      .instance()
      .handleSubmit({preventDefault: () => {}})
      .then(() => {
        expect(wrapper.instance().state.errors.nonField).toEqual("errors");
        expect(wrapper.find(".error.non-field")).toHaveLength(1);
        expect(lastConsoleOutuput).not.toBe(null);
        expect(spyToastError.mock.calls.length).toBe(1);
        expect(spyToastSuccess.mock.calls.length).toBe(0);
        lastConsoleOutuput = null;
      })
      .then(() =>
        wrapper
          .instance()
          .handleSubmit({preventDefault: () => {}})
          .then(() => {
            expect(wrapper.instance().state.errors.nonField).toEqual(
              "non field errors",
            );
            expect(lastConsoleOutuput).not.toBe(null);
            expect(spyToastError.mock.calls.length).toBe(2);
            expect(spyToastSuccess.mock.calls.length).toBe(0);
            lastConsoleOutuput = null;
          }),
      )
      .then(() =>
        wrapper
          .instance()
          .handleSubmit({preventDefault: () => {}})
          .then(() => {
            expect(wrapper.instance().state.errors.nonField).toEqual(
              "token: Invalid token",
            );
            expect(lastConsoleOutuput).not.toBe(null);
            expect(spyToastError.mock.calls.length).toBe(3);
            expect(spyToastSuccess.mock.calls.length).toBe(0);
            lastConsoleOutuput = null;
          }),
      )
      .then(() =>
        wrapper
          .instance()
          .handleSubmit({preventDefault: () => {}})
          .then(() => {
            expect(wrapper.instance().state.errors).toEqual({});
            expect(wrapper.instance().state.success).toBe(true);
            expect(wrapper.find(".input.error")).toHaveLength(0);
            expect(wrapper.find(".success")).toHaveLength(1);
            expect(lastConsoleOutuput).toBe(null);
            expect(spyToastError.mock.calls.length).toBe(3);
            expect(spyToastSuccess.mock.calls.length).toBe(1);
            lastConsoleOutuput = null;
          }),
      );
  });
  it("should set title", () => {
    const setTitleMock = wrapper.instance().props.setTitle.mock;
    expect(setTitleMock.calls.pop()).toEqual(["Reset Password", props.orgName]);
  });
  it("should toggle password icon for both password fields in PasswordToggleIcon", async () => {
    const nodes = wrapper.find(PasswordToggleIcon);
    expect(nodes.length).toEqual(2);
    expect(nodes.at(0).props()).toEqual({
      hidePassword: true,
      inputRef: {current: null},
      isVisible: false,
      parentClassName: "",
      secondInputRef: {current: null},
      toggler: expect.any(Function),
    });
    expect(wrapper.instance().state.hidePassword).toEqual(true);
    nodes.at(0).props().toggler();
    expect(wrapper.instance().state.hidePassword).toEqual(false);
    expect(nodes.at(1).props()).toEqual({
      hidePassword: true,
      inputRef: {current: null},
      isVisible: false,
      parentClassName: "",
      secondInputRef: {current: null},
      toggler: expect.any(Function),
    });
    nodes.at(1).props().toggler();
    expect(wrapper.instance().state.hidePassword).toEqual(false);
  });
});
