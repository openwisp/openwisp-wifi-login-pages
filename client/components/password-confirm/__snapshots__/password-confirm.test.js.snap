// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PasswordConfirm /> rendering should render correctly 1`] = `
<div
  className="container content"
  id="password-confirm"
>
  <div
    className="inner"
  >
    <form
      className="main-column"
      onSubmit={[Function]}
    >
      <div
        className="inner"
      >
        <h1>
          Reset your password
        </h1>
        <p>
          Please enter your new password below.
        </p>
        <div
          className="fieldset"
        >
          <div
            className="row password"
          >
            <label
              htmlFor="password"
            >
              Password
            </label>
            <input
              autoComplete="new-password"
              className="input "
              id="password"
              name="newPassword1"
              onChange={[Function]}
              pattern=".{6,}"
              placeholder="password"
              required={true}
              title="password must be a minimum of 6 characters"
              type="password"
              value=""
            />
            <div
              className=""
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex={0}
            >
              <i
                className="eye"
                title="reveal password"
              />
            </div>
          </div>
          <div
            className="row password-confirm"
          >
            <label
              htmlFor="password-confirm"
            >
              Confirm password
            </label>
            <input
              autoComplete="new-password"
              className="input "
              id="password-confirm"
              name="newPassword2"
              onChange={[Function]}
              pattern=".{6,}"
              placeholder="confirm password"
              required={true}
              title="password must be a minimum of 6 characters"
              type="password"
              value=""
            />
            <div
              className=""
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex={0}
            >
              <i
                className="eye"
                title="reveal password"
              />
            </div>
          </div>
        </div>
        <div
          className="row submit"
        >
          <input
            className="button full"
            type="submit"
            value="Change password"
          />
        </div>
        <div
          className="contact-us"
        >
          Please contact us if you have any trouble resetting your password.
        </div>
        <a
          className="link"
          href="/default/login"
          onClick={[Function]}
        >
          Go back to sign in
        </a>
      </div>
    </form>
    <div
      className="side-column contact"
    >
      <div
        className="inner"
      >
        
        <div
          className="row"
        >
          <span
            className="label"
          >
            Email
            :
          </span>
          <a
            className="link"
            href="mailto:support@openwisp.co"
          >
            support@openwisp.co
          </a>
        </div>
        <div
          className="row"
        >
          <span
            className="label"
          >
            Helpdesk
            :
          </span>
          <a
            className="link"
            href="tel:+789 948 564"
          >
            +789 948 564
          </a>
        </div>
        <div
          className="contact-links"
        >
          <a
            className="link "
            href="https://twitter.com/openwisp"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="twitter"
              className="contact-image "
              src="/assets/default/twitter.svg"
            />
          </a>
          <a
            className="link "
            href="https://facebook.com/openwisp"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="facebook"
              className="contact-image "
              src="/assets/default/facebook.svg"
            />
          </a>
        </div>
        
      </div>
    </div>
  </div>
</div>
`;

exports[`<PasswordConfirm /> rendering with placeholder translation tags should render translation placeholder correctly 1`] = `
<div
  className="container content"
  id="password-confirm"
>
  <div
    className="inner"
  >
    <form
      className="main-column"
      onSubmit={[Function]}
    >
      <div
        className="inner"
      >
        <h1>
          PWD_CONFIRM_H
        </h1>
        <p>
          PWD_CONFIRM_ADD_TXT
        </p>
        <div
          className="fieldset"
        >
          <div
            className="row password"
          >
            <label
              htmlFor="password"
            >
              PWD_LBL
            </label>
            <input
              autoComplete="new-password"
              className="input "
              id="password"
              name="newPassword1"
              onChange={[Function]}
              pattern=".{6,}"
              placeholder="PWD_PHOLD"
              required={true}
              title="PWD_PTRN_DESC"
              type="password"
              value=""
            />
            <PasswordToggleIcon
              hidePassword={true}
              inputRef={
                Object {
                  "current": null,
                }
              }
              isVisible={false}
              parentClassName=""
              secondInputRef={
                Object {
                  "current": null,
                }
              }
              toggler={[Function]}
            />
          </div>
          <div
            className="row password-confirm"
          >
            <label
              htmlFor="password-confirm"
            >
              CONFIRM_PWD_LBL
            </label>
            <input
              autoComplete="new-password"
              className="input "
              id="password-confirm"
              name="newPassword2"
              onChange={[Function]}
              pattern=".{6,}"
              placeholder="CONFIRM_PWD_PHOLD"
              required={true}
              title="PWD_PTRN_DESC"
              type="password"
              value=""
            />
            <PasswordToggleIcon
              hidePassword={true}
              inputRef={
                Object {
                  "current": null,
                }
              }
              isVisible={false}
              parentClassName=""
              secondInputRef={
                Object {
                  "current": null,
                }
              }
              toggler={[Function]}
            />
          </div>
        </div>
        <div
          className="row submit"
        >
          <input
            className="button full"
            type="submit"
            value="PWD_CONFIRM"
          />
        </div>
        <div
          className="contact-us"
        >
          PWD_RESET_CNTC_TXT
        </div>
        <Link
          className="link"
          to="/default/login"
        >
          LOGIN_PG_LNK
        </Link>
      </div>
    </form>
    <Connect(Contact) />
  </div>
</div>
`;
